### ISSUE TITLE
to_hierarchical_dataframe() DataFrames are too wide and long

### ISSUE THREAD

**Issue by lvsltz** [oruebel/ndx-icephys-meta#23]
Not sure if in your control but I mention this anyway.
`to_hierarchical_dataframe()` and `to_denormalized_dataframe()` generate very wide and long DataFrames. To save some space and facilitate visualization, the `stimulus`, `response` and `electrode` columns from `intracellular_recordings` should contain fewer details.

![image](https://user-images.githubusercontent.com/6717189/69877231-e947b400-12c1-11ea-952c-4a0b2d4081f9.png)

**Comment by oruebel** [comment]
@lvsltz that makes sense. I think the proper solution would be to wrap the ``stimulus`` and ``response`` columns in a separate class (and to give them a new neurodata_type) so that we can control the representation. That is certainly possible but requires a bit of coding and changes to schema, i.e., it may take a bit before I can get to this issue.

**Comment by oruebel** [comment]
A simple workaround would be to set the maximum width for columns in the pandas display, e.g.,:
```
import pandas
pandas.set_option("display.max_colwidth", 30)
```

This allows you to control the column width in the rendering. However, it would still be nice to have nicer text representations for the ``stimulus`` and ``response`` columns.

**Comment by lvsltz** [comment]
Yes, this can work too. What I was thinking of is more to decorate python's `display()` or hdmf's `Container` when a denormalized_dataframe is shown, like below:

```python
from pynwb.icephys import PatchClampSeries, IntracellularElectrode

def newrepr(self):
    cls = self.__class__
    if cls.__name__ in ['VoltageClampStimulusSeries',
                        'VoltageClampSeries',
                        'CurrentClampStimulusSeries',
                        'CurrentClampSeries',
                        'IntracellularElectrode']:
        template = "%s" % (self.name)
    else:
        template = self.__repr__()
    return template

IntracellularElectrode.__repr__ = newrepr
PatchClampSeries.__repr__ = newrepr
```
![image](https://user-images.githubusercontent.com/6717189/76076367-0534b300-5f9f-11ea-9ee5-b56459590dbc.png)

But this issue is really not very important, you can even close it :)
